File: src/main/py/rckpprsci.py
# Rock-Paper-Scissors
# program that lets the user play Rock-Paper-Scissors against the computer
from random import randint


def rock_paper_scissors():
    print("program that lets the user play Rock-Paper-Scissors against the computer")
    rock = 1
    paper = 2
    scissor = 3
    # paper > rock > scissor > paper

    user_wins = 0
    computer_wins = 0
    print("scoreboard:: user:",user_wins,"computer:",computer_wins)
    draws = 0
    exit_entered = False
    while not exit_entered:
        compPick = randint(1,3)
        uPick = input("enter your number: 1.rock, 2.paper, 3.scissor (or 'exit' to quit): ")
        userPick = eval(uPick) if uPick.isnumeric() else 0
        if ((compPick == rock and userPick == rock)
            or (compPick == paper and userPick == paper)
            or (compPick == scissor and userPick == scissor)):
                draws = draws + 1
        elif ((compPick == rock and userPick == paper)
            or (compPick == paper and userPick == scissor)
            or (compPick == scissor and userPick == rock)):
                user_wins = user_wins + 1
        elif ((compPick == rock and userPick == scissor)
            or (compPick == paper and userPick == rock)
            or (compPick == scissor and userPick == paper)):
                computer_wins = computer_wins + 1
        print("scoreboard:: user:",user_wins,"computer:",computer_wins)
        if uPick == "exit":
            exit_entered = True

    print("Results -- user wins",user_wins,"times, computer wins",computer_wins,"times, and draws",draws,"times")
    pass


rock_paper_scissors()

File: src/main/java/rckpprsci.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class rckpprsci {
    private static final int ROCK = 1;
    private static final int PAPER = 2;
    private static final int SCISSORS = 3;
    private static Map<String, Integer> score;

    public static void main(String[] args) {
        playRockPaperScissors();
    }

    public static void playRockPaperScissors() {
        resetScore();
        boolean exit_entered = Boolean.FALSE;
        try (Scanner kbReader = new Scanner(System.in)) {
            do {
                System.out.println("enter your number: 1.rock, 2.paper, 3.scissor (or 'exit' to quit): ");
                String usersChoice = kbReader.next();
                if (usersChoice.equalsIgnoreCase("exit")) {
                    exit_entered = Boolean.TRUE;
                    break;
                }
                int userChoice = Integer.parseInt(usersChoice);
                int computerChoice = getComputerChoice();

                // score calculation
                trackScore(userChoice, computerChoice);
                displayChoices(userChoice, computerChoice);
            } while (!exit_entered);
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }

        System.out.println(
            "Final Results -- user wins "+score.get("user_wins")
            + " times, computer wins "+score.get("computer_wins")
            + " times, and draws "+score.get("draws")+" times"
            );
    }

    private static void displayChoices(int userChoice, int computerChoice) {
        String uVal, cVal;
        if (userChoice == ROCK) {
            uVal = "rock";
        } else if (userChoice == PAPER) {
            uVal = "paper";
        } else {
            uVal = "scissors";
        }
        if (computerChoice == ROCK) {
            cVal = "rock";
        } else if (computerChoice == PAPER) {
            cVal = "paper";
        } else {
            cVal = "scissors";
        }
        System.out.println("userChoice: " + uVal + ", computerChoice: " + cVal);
    }

    private static int getComputerChoice() {
        return (int) (Math.random() * 3) + 1;
    }   

    private static void resetScore() {
        score = new HashMap<>();
        score.put("user_wins", 0);
        score.put("computer_wins", 0);
        score.put("draws", 0);
    }

    private static void trackScore(int userChoice, int computerChoice) {
        if (userChoice == computerChoice) {
            score.put("draws", score.get("draws") + 1);
        } else if ((userChoice == ROCK && computerChoice == SCISSORS)
                || (userChoice == PAPER && computerChoice == ROCK)
                || (userChoice == SCISSORS && computerChoice == PAPER)) {
            score.put("user_wins", score.get("user_wins") + 1);
        } else if ((userChoice == SCISSORS && computerChoice == ROCK)
                || (userChoice == ROCK && computerChoice == PAPER)
                || (userChoice == PAPER && computerChoice == SCISSORS)) {
            score.put("computer_wins", score.get("computer_wins") + 1);
        }
    }

    
}